save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -34,280 +36,192 @@ import react.dom.html.ReactHTML.small
 import react.dom.html.ReactHTML.span
 import react.dom.html.ReactHTML.ul
 import react.router.useLocation
+import react.router.useNavigate
 import react.useState
 
-import kotlinx.browser.window
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.isActive
 
 /**
- * [Props] of the top bor component
+ * Displays the url and its division by "/"
  */
-external interface TopBarProps : PropsWithChildren {
-    /**
-     * Currently logged in user or null
-     */
-    var userInfo: UserInfo?
-
-    /**
-     * true if the device is mobile (screen is less 1000px)
-     */
-    var isMobileScreen: Boolean?
+private val topBarUrlSplits: FC<TopBarPropsWithLocation> = FC { props ->
+    nav {
+        className = ClassName("navbar-nav mr-auto w-100")
+        ariaLabel = "breadcrumb"
+        ol {
+            className = ClassName("breadcrumb mb-0")
+            li {
+                className = ClassName("breadcrumb-item")
+                ariaCurrent = "page".unsafeCast<AriaCurrent>()
+                a {
+                    href = "#/"
+                    // if we are on welcome page right now - need to highlight SAVE in menu
+                    val textColor = if (props.location.pathname == "/") "text-warning" else "text-light"
+                    className = ClassName(textColor)
+                    +"SAVE"
+                }
+            }
+            props.location.pathname
+                .substringBeforeLast("?")
+                .split(URL_PATH_DELIMITER)
+                .filterNot { it.isBlank() }
+                .apply {
+                    val url = TopBarUrl(props.location.pathname.substringBeforeLast("?"))
+                    forEachIndexed { index: Int, pathPart: String ->
+                        url.changeUrlBeforeButton(pathPart)
+                        if (url.isCreateButton(index)) {
+                            li {
+                                className = ClassName("breadcrumb-item")
+                                ariaCurrent = "page".unsafeCast<AriaCurrent>()
+                                if (index == size - 1) {
+                                    a {
```

Comment:
```kotlin
Link {
   to = /*...*/
}
```

Reply:
There seems to be no link here, and with Link, without using to, pages are not rendered. Therefore, this case left a {}, the rest was redone with Link {}

Reply:
Are you sure you have tried everything from `react.router.dom`? Either you do something wrong or you haven't tried the suitable component :)

Reply:
N/A in this particular case.